packages/connection/src/class-plugin-storage.php

```diff
@@ -116,11 +137,55 @@ public static function delete( $slug ) {
 	 * @return bool|WP_Error
 	 */
 	private static function ensure_configured() {
-		if ( class_exists( Config::class ) && method_exists( Config::class, 'is_configured' ) && ! Config::is_configured() ) {
+		if ( ! self::$configured ) {
 			return new WP_Error( 'too_early', __( 'You cannot call this method until Jetpack Config is configured', 'jetpack' ) );
 		}
 
 		return true;
 	}
 
+	/**
+	 * Called once to configure this class after plugins_loaded.
+	 *
+	 * @return void
+	 */
+	public static function configure() {
+
+		if ( self::$configured ) {
+			return;
+		}
+
+		// If a plugin was activated or deactivated.
+		$number_of_plugins_differ = count( self::$plugins ) !== count( get_option( self::ACTIVE_PLUGINS_OPTION_NAME, array() ) );
+
+		if ( $number_of_plugins_differ || true === self::$refresh_connected_plugins ) {
+			self::update_active_plugins_option();
+		}
+
+		add_filter( 'jetpack_sync_options_whitelist', __NAMESPACE__ . '\Plugin_Storage::whitelist_options' );
+
+		self::$configured = true;
+
+	}
+
+	/**
+	 * Adds Plugins Storage options to sync whitelist
+	 *
+	 * @param array $options the whitelisted options.
+	 * @return array
+	 */
+	public static function whitelist_options( $options ) {
+		$options[] = self::ACTIVE_PLUGINS_OPTION_NAME;
+		return $options;
+	}
+
+	/**
+	 * Updates the active plugins option with current list of active plugins.
+	 *
+	 * @return void
+	 */
+	public static function update_active_plugins_option() {
+		update_option( self::ACTIVE_PLUGINS_OPTION_NAME, array_keys( self::$plugins ) );
```

Comment:
Using `update_option` directly may lead to a bunch of issues on network sites.
Why have you decided against using `Jetpack_Options`, which takes care of that automatically?

Reply:
It's not decided, see the `TODO` section on the PR description

Reply:
I'd recommend you to go ahead with it, worked as a charm for me (see PR #15669).